### DISCUSSION TITLE
Should the "controls" update the story immediately?

### DISCUSSION THREAD

**Discussion by trajano** [storybookjs/react-native#642] [Q&A] [answered]
I am not sure if I am doing this right, but I was expecting that if I have controls like `fontSize` and I set it to 40 on the control view it should update the story, but it seems I have to leave the story and come back for it to take effect.  Am I missing something?

**Comment by dannyhw** [comment]
that shouldn't be the case, I don't see that happening in my testing, could you show a gif or something? A reproduction would also be appreciated.

**Reply by trajano** [reply]
It's in my experiments I don't think there's anything really special.

https://github.com/trajano/expo-experiments/blob/e7d6e66b1a53b414750fc0ce7e6c137e8374c0d9/packages/my-app/src/stories/Fonts.stories.tsx

Note the `boolean` does toggle just the weights do not update.  Now that I think about it it may be a flashlist "key" thing I may need to handle

**Comment by trajano** [comment, marked as answer]
It appears to be an open bug/feature of flashlist https://github.com/Shopify/flash-list/issues/699 adding a key extractor fixed the issue
